### ISSUE TITLE
[BUG] myloader hangs (current master version)

### ISSUE THREAD

**Issue by LinuxJedi** [mydumper/mydumper#915]
**Describe the bug**
Current master of myloader hangs during restore (not attempted to diagnose the threads yet).

**To Reproduce**
With a basic MariaDB 10.3 server and Sakila test data:
Command executed:
* mydumper --user=root --socket=/tmp/mysql.sock
* myloader --verbose=4 --user=root --socket=/tmp/mysql.sock --directory=export-20221123-153402/ --source-db=sakila

What mydumper and myloader version has been used?

0.13.0-6 (master head)

**Expected behavior**
Don't hang

**Environment (please complete the following information):**
 - OS version: Ubuntu 22.04

**Comment by LinuxJedi** [comment]
I've also had reports that this happens in 0.12 as well, but I haven't been able to reproduce it in this yet. Sakila data set is likely too small and I haven't tried ramping the threads up yet.

**Comment by LinuxJedi** [comment]
OK, so a bit more information here. In 0.13 this happens due to the `process_stream_queue`, as soon as "Intermediate thread: SHUTDOWN" is hit all execution is halted.

**Comment by LinuxJedi** [comment]
#936 didn't fix this for me.

**Comment by billynoah** [comment]
Is this bug back?  I am getting this issue with latest v0.16.1-3.

```
** Message: 11:10:32.400: Using 2 loader threads
** Message: 11:10:32.400: Merging config files user:
** Message: 11:10:32.403: Connection via default library settings
** Message: 11:10:32.404: Initializing initialize_worker_schema
** Message: 11:10:32.405: S-Thread 4: Starting import
** Message: 11:10:32.405: S-Thread 3: Starting import
** Message: 11:10:32.405: S-Thread 6: Starting import
** Message: 11:10:32.405: S-Thread 5: Starting import
** Message: 11:10:32.406: Reading metadata: metadata
** Message: 11:10:32.423: Intermediate queue: Sending END job
** Message: 11:10:32.423: Intermediate thread ended
** Message: 11:10:32.423: Intermediate thread: SHUTDOWN
```

**Comment by JuttuJ** [comment]
Same here.

```
025-10-27 18:26:16 [INFO] - Fast index creation will be used for table: xxx.xxx
2025-10-27 18:26:16 [INFO] - Intermediate thread ended
2025-10-27 18:26:16 [INFO] - Intermediate thread: SHUTDOWN
```

**Comment by davidducos** [comment]
Hi @JuttuJ, can you be more specific on your case? which version? did you test with the latest one? can you give more info about your backup specifications? amount of files? kind of tables? a full log? a test case?

**Comment by JuttuJ** [comment]
Hi  @davidducos,
sure!

**Version:** MyDumper restore version: 0.20.1-2
**Command:**
```
$ time myloader -h <host> \
-u <user> -a -d <backup_path> -B <database> -t 15 -v 3 \
--drop-table -L <log_path>
```
**Backup size:** 12 GB

Unfortunately I cannot give any other details, because when I waited for a couple of minutes and retried, it worked out of a sudden. I overwrote all of the logs.

When I initially started **myloader** with the above comment, then I stopped it using **CTRL +C** and waited for it to shut down. I have removed the `resume` file in the backup directory and tried to restart and then I saw this error message.

Best thing to do was to get a drink and then the command worked fine again.
